tests/testthat/test-trace.R

```diff
@@ -10,6 +10,32 @@ public({
     expect_error(stop_capturing(), NA)
     expect_error(stop_capturing(), NA)
   })
+
+  FUNS <- list(
+    httr = c('request_perform', 'body_config'),
+    curl = c('form_file')
+  )
+  
+  .are_pkgs_traced <- function() {
+    .process_fun <- function(fun, pkg) {
+      inherits(getFromNamespace(fun, pkg), "functionWithTrace")
+    }
+    .process_pkg <- function(pkg) {
+      sapply(FUNS[[pkg]], .process_fun, pkg = pkg)
+    }
+    unlist(lapply(names(FUNS), .process_pkg))
+  }
+
+  with_mock_api({
+    test_that("curl/httr functions are properly traced", {
+      expect_true(all(.are_pkgs_traced()))
+    })
+  })
+
+  test_that("curl/httr functions are properly untraced", {
+      expect_true(all(!.are_pkgs_traced()))
```

Comment:
Same logic but I think the assertion reads more intuitively this way

```suggestion
      expect_false(any(.are_pkgs_traced()))
```